lib-functional/index.js

```diff
@@ -0,0 +1,143 @@
+function applyParsingFunctions(markdownFile) {
+  return functionNames.reduce(function(markdownFileState, functionName) {
+    return functionName(markdownFileState)
+  }, markdownFile)
+}
+
+var functionNames = [addImages, addLinks, addCodeBlock, removeCodeBlockMD,
+  addInlineCode, removeInlineCodeMD, addBlockQuotes, removeBlockQuotesMD,
+  addH6, addH5, addH4, addH3, addH2, addH1, removeHeadersMD, addHorizontalRule,
+  removeHorizontalMD, addOrderedList, addUnorderedList, addListItems,
+  removeListItemsMD, addBoldHTML, removeBoldMD, addItalicsHTML, removeItalicsMD,
+  addParagraphHTML]
+
+exports.applyParsingFunctions = applyParsingFunctions
+
+
+/** PARSING FUNCTIONS **/
+
+// BOLD //
```

Comment:
I think you can get rid of all these comments. The function names are explicit enough.